freqtrade/freqtradebot.py

```diff
@@ -756,23 +755,28 @@ def _check_and_execute_sell(self, trade: Trade, sell_rate: float,
             return True
         return False
 
+    def _check_timed_out(self, side: str, order: dict) -> bool:
+        """
+        Check if timeout is active, and if the order is still open and timed out
+        """
+        timeout = self.config.get('unfilledtimeout', {}).get(side)
+        ordertime = arrow.get(order['datetime']).datetime
+        if timeout:
```

Comment:
This 1) won't actually happen (the `timeout==0` case) since limits for those values in the config schema are positive; 2) probably changes the behavior of the function here (previous code does not contain this check); 3) if left (it may have sense, to have this function working as a box, without knowledge on limits applies to the settings in the config schema), this should probably be changed to `if timeout > 0:`

Reply:
1) yes, currently it can't happen - but it may happen in the future, assuming we allow loading this from the strategy - so we could get an incomplete dict. 
see #2513 for details.

The check was also there before - but outside of this section (checking if "unfilledtimeout" is in the config must be seen equal to this check.

>  this should probably be changed to `if timeout > 0:`

hmm why disallow a limit of 0? it can simulate a "FOK" / "IOK" order on exchanges which don't support this.

Reply:
> The check was also there before - but outside of this section (checking if "unfilledtimeout" is in the config must be seen equal to this check.

No, now if timeout is 0, the `timeout_threshold = arrow.utcnow().shift(minutes=-timeout).datetime` won't be calculated as well as further calculation for `(order['status'] == 'open' and order['side'] == side and ordertime < timeout_threshold)` will not be performed and False will be returned. Previously, if `buy_timeout = self.config['unfilledtimeout']['buy']` is set to 0 (well, assume it's from strategy, as you say), the `buy_timeout_threshold = arrow.utcnow().shift(minutes=-buy_timeout).datetime` and further check for `or (order['status'] == 'open' and order['side'] == 'buy' and ordertime < buy_timeout_threshold)` were performed anyway. That's the difference in behavior, specifically for the case when timeout value is 0.

> hmm why disallow a limit of 0? it can simulate a "FOK" / "IOK" order on exchanges which don't support this.

But you've disallowed it with that check `if timeout:` anyway... 😉

Reply:
ah yes - so the correct check should be `if timeout is None:` ?

Reply:
I do not know, both cannot be set in the config in the meantime...
What's the intention of this check?

Reply:
a) prepare to support 0 (The minimums seem very randomly chosen anyway) ... 
b) pepare to have `"unfilledtimeout"` only available for one of the two 
c) point b this **is** possible at the moment - none of the 2 keys are mandatory - so my config can look as follows and be valid.

```
"unfilledtimeout": {
        "buy": 10,
    },

```

This will crash (in develop) once it has the first open sell-order - but it's still a valid configuration - so this check will (implicitly) support this type of configuration.